src/utils/IntPoint.h

```diff
@@ -55,8 +55,8 @@ static Point no_point(std::numeric_limits<ClipperLib::cInt>::min(), std::numeric
 INLINE Point operator-(const Point& p0) { return Point(-p0.X, -p0.Y); }
 INLINE Point operator+(const Point& p0, const Point& p1) { return Point(p0.X+p1.X, p0.Y+p1.Y); }
 INLINE Point operator-(const Point& p0, const Point& p1) { return Point(p0.X-p1.X, p0.Y-p1.Y); }
-template<typename T> INLINE Point operator*(const Point& p0, const T i) { return Point(p0.X * i, p0.Y * i); }
```

Comment:
The reason that this needed to be changed is because there was an ambiguity because T could be anything, including `Point`s. The constructor for `IntPoint` in ClipperLib has default values for its `x` and `y` parameters so that `Point()` creates a point at coordinate 0,0. But this has the side-effect that you could also do `Point(5)` and then create a point at coordinate 5,0. That then allows a numerical constant to be cast into a `Point` and the ambiguity was that `Point(3, 4) * 5` could evaluate to `operator *(Point(3, 4), 5)` or to `operator *(Point(3, 4), Point(5, 0)`.

The only place where this operator was used, was multiplying `Point * coord_t`. It then knows that `coord_t` is a special case and always prefers that rather than casting `coord_t` to a `Point` first.

Reply:
Yeah, that template specialisation is nice. Implemented: 8fa9e738eb721deab201356c8ac8d9ebbaec2f65